vm_eval.c

```diff
@@ -1613,7 +1613,23 @@ rb_each(VALUE obj)
     return rb_call(obj, idEach, 0, 0, CALL_FCALL);
 }
 
-static VALUE eval_default_path;
+static VALUE eval_default_path = Qfalse;
+
+static VALUE
+get_eval_default_path(void)
+{
+    int location_lineno;
+    VALUE location_path = rb_source_location(&location_lineno);
+    if (!NIL_P(location_path)) {
+        return rb_fstring(rb_sprintf("(eval at %s:%d)", RSTRING_PTR(location_path), location_lineno));
```

Comment:
Preserve the encoding.

```suggestion
        return rb_fstring(rb_sprintf("(eval at %"PRIsVALUE":%d)", location_path, location_lineno));
```